the-collab-lab/tcl-45-smart-shopping-list
# copy feature available using react hot toast and navigator.clipboard

## Description

This PR has a button that allows a user to copy the listToken (list name) onto clipboard. It uses react hot toast to notify the user that a name was copied successfully, and the button changes from "Copy List Name" to "Copied!" for 2 seconds. I am open to changing it for better accessibility, but for now the functionality of copying it onto clipboard should be working. I also assumed that by "sharing list feature" we just wanted a way to copy it onto the clipboard easily instead of highlighting + c + v. We can style it better in the `List` styling issue. 

## Related Issue

Closes #39 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

## Testing Steps / QA Criteria

1. From your local machine git checkout main then git pull
2. git checkout MK-copy-list-feature
3. npm run start
4. Go to `List` and click on "Copy List Name"
5. A pop up should come up saying "Copied successfully" and the button should change from "Copy List Name" to "Copied!"
6. Go back to Home after logging out, and try out the newly copied list name that is on your clipboard. You can also just paste it somewhere to see that it was copied onto your machine.
